drivers/pwm/pwm_nrfx.c

```diff
@@ -376,6 +382,12 @@ static int pwm_nrfx_pm_action(const struct device *dev,
 			 POST_KERNEL, CONFIG_PWM_INIT_PRIORITY,		      \
 			 &pwm_nrfx_drv_api_funcs)
 
+#define PWM_MEMORY_SECTION(idx)						      \
```

Comment:
Why not define it before `PWM_NRFX_DEVICE()` where it is used?

Reply:
I have done it analogously to SPIM driver: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/spi/spi_nrfx_spim.c#L685 to preserve consistency. If you find it more readable I can relocate it

Reply:
Yes, please relocate this. I don't know why it is like that in `spi_nrfx_spim.c`. It feels natural to have the macro definition before its use.

Reply:
relocated